buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantCommandTask.groovy

```diff
@@ -32,21 +34,37 @@ import javax.inject.Inject
 public class VagrantCommandTask extends LoggedExec {
 
     @Input
+    String command
+
+    @Input @Optional
+    String subcommand
+
+    @Input
     String boxName
 
     @Input
     Map<String, String> environmentVars
 
     public VagrantCommandTask() {
         executable = 'vagrant'
+    }
 
-        project.afterEvaluate {
-            // It'd be nice if --machine-readable were, well, nice
-            standardOutput = new TeeOutputStream(standardOutput, createLoggerOutputStream())
-            if (environmentVars != null) {
-                environment environmentVars
-            }
+    @Override @TaskAction
+    protected void exec() {
+        // Build our command line for vagrant
+        def vagrantCommand = [executable, command]
+        if (subcommand != null) {
+            vagrantCommand + subcommand
         }
+        commandLine([*vagrantCommand, boxName, *args])
+
+        // It'd be nice if --machine-readable were, well, nice
+        standardOutput = new TeeOutputStream(standardOutput, createLoggerOutputStream())
+        if (environmentVars != null) {
+            environment environmentVars
+        }
+
+        super.exec()
```

Comment:
Ironically, to do this it seems that it needs to be called from within a `project.afterEvaluate` call. I'm finding out that calling `doFirst` from within the constructor still slots the main task action in front of the closure given in the constructor. I feel like that's an appropriate use of `afterEvaluate`, but before I do, I wanted to check to see if there's a better way.